src/Application/Actions/Donations/Update.php

```diff
@@ -374,7 +375,18 @@ private function addData(
                     /** @var string|null $nextActionType */
                     $nextActionType = null;
                     if ($confirmedIntent->status === PaymentIntent::STATUS_REQUIRES_ACTION) {
-                        $nextActionType = (string) $confirmedIntent->next_action?->type;
+                        /** @var ?ErrorObject $nextAction */
+                        $nextAction = $confirmedIntent->next_action;
+
+                        if ($nextAction && ! $nextAction instanceof ErrorObject) {
```

Comment:
How do we know this is an `ErrorObject`?

If it is, it seems like our logic deviates from the docs:
* `ErrorObject::$type` is annotated "One of `api_error`, `card_error`, `idempotency_error`, or `invalid_request_error`."
* Just after this change, we are using the type on the left side of `$nextActionType === 'display_bank_transfer_instructions'`

Reply:
I wasn't 100% sure that it would be an ErrorObject, which is why I made it log an error but not necessarily stop if its a different type. I guess it probably isn't an ErrorObject then. Do you have an idea what else it would be? If we just remove the @var annotation we get this Psalm error:

```

ERROR: UndefinedMagicPropertyFetch
at /home/barney/projects/matchbot/src/Application/Actions/Donations/Update.php:389:52
Magic instance property Stripe\StripeObject::$type is not defined (see https://psalm.dev/218)
                        $nextActionType = (string) $nextAction?->type;
```

Reply:
Replaced with a Psalm suppress in 7b3a04b6383f8